bam-cmd/src/main/java/org/victorchang/QnameCommand.java

```diff
@@ -129,14 +135,46 @@ private boolean checkDirectory(Path path) {
     }
 }
 
+class QnameParam {
+    @Option(names = {"-n", "--name"}, description = "QNAME to search for")
+    String qname;
+    @Option(names = {"-f", "--file-name"}, description = "Path to a file containing QNAMEs to search for (separated by newline).")
+    Path qnamePath;
+
+    List<String> getQnames() {
+        if (qname != null) {
+            return singletonList(qname);
+        }
+        try {
+            return Files.readAllLines(qnamePath)
+                    .stream()
+                    .filter(this::isValidQname)
+                    .collect(toList());
+        } catch (Exception e) {
+            LOG.error("Could not read file : {}", qnamePath, e);
+            return emptyList();
+        }
+    }
+
+    private boolean isValidQname(String qName) {
+        try {
+            SAMSequenceRecord.validateSequenceName(qName);
+            return true;
+        } catch (Exception e) {
+            return false;
```

Comment:
I think we should error in this case. We don't want users to think "This QNAME is not in this file" when in fact they might have copied it wrong or accidentally added a character or something.